stratosphere/pm/source/impl/pm_process_info.hpp

```diff
@@ -0,0 +1,310 @@
+ *
+ * This program is distributed in the hope it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+#include <switch.h>
+#include <stratosphere.hpp>
+#include <stratosphere/ldr.hpp>
+#include <stratosphere/pm.hpp>
+
+#include "pm_process_manager.hpp"
+
+namespace sts::pm::impl {
+
+    class ProcessInfo {
+        NON_COPYABLE(ProcessInfo);
+        private:
+            enum Flag : u32 {
+                Flag_SignalOnExit           = (1 << 0),
+                Flag_ExceptionOccurred      = (1 << 1),
+                Flag_ExceptionWaitingAttach = (1 << 2),
+                Flag_SignalOnDebugEvent     = (1 << 3),
+                Flag_SuspendedStateChanged  = (1 << 4),
+                Flag_Suspended              = (1 << 5),
+                Flag_Application            = (1 << 6),
+                Flag_SignalOnStart          = (1 << 7),
+                Flag_StartedStateChanged    = (1 << 8),
+            };
+        private:
+            const u64 process_id;
+            const ldr::PinId pin_id;
+            const ncm::TitleLocation loc;
+            Handle handle;
+            ProcessState state;
+            u32 flags;
+        private:
+            void SetFlag(Flag flag) {
+                this->flags |= flag;
+            }
+
+            void ClearFlag(Flag flag) {
+                this->flags &= ~flag;
+            }
+
+            bool HasFlag(Flag flag) const {
+                return (this->flags & flag);
+            }
+        public:
+            ProcessInfo(Handle h, u64 pid, ldr::PinId pin, const ncm::TitleLocation &l);
+            ~ProcessInfo();
+            void Cleanup();
+
+            Handle GetHandle() const {
+                return this->handle;
+            }
+
+            u64 GetProcessId() const {
+                return this->process_id;
+            }
+
+            ldr::PinId GetPinId() const {
+                return this->pin_id;
+            }
+
+            const ncm::TitleLocation &GetTitleLocation() {
+                return this->loc;
+            }
+
+            ProcessState GetState() const {
+                return this->state;
+            }
+
+            void SetState(ProcessState state) {
+                this->state = state;
+            }
+
+            void SetSignalOnExit() {
+                this->SetFlag(Flag_SignalOnExit);
+            }
+
+            void SetExceptionOccurred() {
+                this->SetFlag(Flag_ExceptionOccurred);
+                this->SetFlag(Flag_ExceptionWaitingAttach);
+            }
+
+            void ClearExceptionOccurred() {
+                this->ClearFlag(Flag_ExceptionOccurred);
+            }
```

Comment:
lots of boilerplate with these flag setters/clearers... macro maybe?

Reply:
Yeah, I thought about it. I'll see how it looks?